core/src/main/java/hudson/model/Run.java

```diff
@@ -1435,39 +1467,60 @@ public String toString() {
         return rawF;
```

Comment:
Consider removing the historical gzip hacks and at best make this into a `LogStorageFactory` that detects `log.gz` existence for compatibility reasons (but never does this for new builds).